### PR TITLE
Add cpplint to Travis

### PR DESCRIPTION
**Pull request by Denaun**

### CODE REVIEW

**Comment by giactra** [comment]
I am from smartphone. What should I click to accept? Close and comment or Merge pull request?

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,14 @@
 sudo: false
 language: cpp
 compiler:
-- clang
+  - clang
 os:
-- osx
+  - osx
+before_install:
+  - wget https://raw.githubusercontent.com/google/styleguide/gh-pages/cpplint/cpplint.py
```
Comment (cristiano-santoni) [inline .travis.yml:8]: Sooo we are going for the google coding style?

Comment (Denaun) [reply .travis.yml:8]: I thought so… Should we vote between Google's and LLVM's?

Comment (cristiano-santoni) [reply .travis.yml:8]: Perhaps, 
in my head I have uniformity fighting against variable names starting with a capital letter...

Comment (Denaun) [reply .travis.yml:8]: Don't know, there are many things that I don't really like in LLVM's style guide…
For functions, I got used to UpperCamelCase, so lowerCamelCase looks weird, plus I like Google's naming for "cheap" functions (snake_case) that also gives some uniformity with the std style guide.
For enums, I don't see why add prefixes when you can just use a strongly typed enum.

Comment (cristiano-santoni) [reply .travis.yml:8]: UpperCamelCase for functions??? are you nuts???
Kidding xD, but seriously, no...
Cheap function named following std convention is clearly an evil thing as it lead to confusion.
On the enum I may agree.
We could keep arguing for a bit a then vote, what do you think?
just to expose points of view.

Comment (Denaun) [reply .travis.yml:8]: No confusion!
```cpp
some_object.BigAndExpensiveMethod()
some_object.simple_method_or_just_a_getter()
some_struct.element_of_the_struct
```
UpperCamelCase for functions is quite common, although it took me a lot to get used to it lol

We can move the discussion to Slack!

**Review by giactra — APPROVED** [review]
Jack; Approved

**Pull request merged**